uber/site_sections/preregistration.py

```diff
@@ -465,6 +466,9 @@ def transfer_badge(self, session, message='', **params):
     def confirm(self, session, message='', return_to='confirm', undoing_extra='', **params):
         attendee = session.attendee(params, restricted=True)
 
```

Comment:
My concern with this pattern is that it forces us to think about all of the places where we don't want an invalid ``Attendee`` to be treated normally and add an if-statement to ensure that the correct page is shown.  In this very file, we probably also want the ``transfer_badge`` page to have a similar check, plus the donation form pages, etc.  I'd also expect a bunch of admin pages to do the same check.

Here's what I'd like to see instead: when we call ``session.attendee(id)``, by default that function checks the attendee badge status, and if it's invalid we get an ``HTTPRedirect`` to ``../preregistration/invalid_badge``.  In cases such as ``/uber/registration/form`` where we explicitly WANT to be able to return an invalid attendee, we can pass an extra param, e.g. ``session.attendee(id, allow_invalid=True)``.

Then we just define that as a separate page here in ``preregistration.py``, e.g.

```python
def invalid_badge(self, session, id):
    return {'attendee': session.attendee(id, allow_invalid=True)}
```

Since ``session.attendee`` is defined automatically in ``models.py`` in ``initialize_db()`` by way of ``_make_getter`` we can just monkeypatch it to add this feature, e.g. we could add something like this to the bottom of ``models.py``:

```python
@on_startup
def _attendee_validity_check():
    orig_getter = Session.SessionMixin.attendee
    @wraps(orig_getter)
    def with_validity_check(self, *args, **kwargs):
        allow_invalid = kwargs.pop('allow_invalid', False)
        attendee = orig_getter(self, *args, **kwargs)
        if not allow_invalid and not attendee.is_new and attendee.badge_status == c.INVALID_STATUS:
            raise HTTPRedirect('../preregistration/invalid_badge?id={}', attendee.id)
        else:
            return attendee
    Session.SessionMixin.attendee = with_validity_check
```

We already do something extremely similar to this in the MIVS plugin, see
https://github.com/magfest/mivs/blob/master/mivs/models.py#L184
for an example of this pattern already being used.